MUXSDKStats/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -312,31 +319,33 @@ - (void)detachAVPlayer {
 }
 
 - (void)monitorAVPlayerItem {
-    if (_playerItem) {
-        [self stopMonitoringAVPlayerItem];
-        [self.playDispatchDelegate videoChangedForPlayer:_name];
-        //
-        // Special case for AVQueuePlayer
-        // In a normal videoChange: world - the KVO for "rate" will fire - and
-        // subsequently after that this binding will dispatchPlay. In fact, any time
-        // an AVPlayer gets an item loaded into it the KVO for "rate" changes.
-        //
-        // However, in AVQueuePlayer world - the "rate" doesn't fire when the video is
-        // changed. I don't know why, but I guess that is the intended behavior. For that
-        // reason, if we're handling a videoChange event and we're dealing with AVQueuePlayer
-        // then we have to fire the play event here.
-        //
-        if (_shouldHandleAVQueuePlayerItem) {
-            _shouldHandleAVQueuePlayerItem = false;
-            [self dispatchPlay];
+    if (_automaticVideoChange) {
```

Comment:
One thing that can help with too much nesting, especially for `void` return signatures, is to bail out directly, rather than nest further. Where possible, I tend to try to _not_ nesting, so do something like the following instead:

```objc
if (!_automaticVideoChange) {
  return;
}

if (_playerItem) {
  ...
```

Reply:
This is a good shout - i've updated to make it not nested